parsers/wiki_parser.py

```diff
@@ -0,0 +1,291 @@
+    * Add original response text, processed response text and file link to a list as a tuple.
+
+    :param responses_source: Mediawiki source
+    :return: list with tuples of (original_text, processed_text, link).
+    """
+    responses_list = []
+    file_and_text_list = []
+
+    response_regex = re.compile(RESPONSE_REGEX)
+    file_regex = re.compile(FILE_REGEX)
+
+    for response in response_regex.finditer(responses_source):
+        original_text = parse_response(response['text'])
+        if original_text == '':
+            pass
+
+        files_source = response['files']
+        for file in file_regex.finditer(files_source):
+            file_name = file['file'].replace('_', ' ').capitalize()
+            file_and_text_list.append([original_text, file_name])
+
+    files_list = [file for text, file in file_and_text_list]
+    file_and_link_dict = links_for_files(files_list)
+
+    for original_text, file in file_and_text_list:
+        processed_text = clean_response_text(original_text)
+        if processed_text != '':
+            try:
+                link = file_and_link_dict[file]
+                responses_list.append((original_text, processed_text, link))
+            except KeyError:
+                # Ignore files with no links to mp3 files. Happens to broken files and files undergoing migration.
+                pass
+
+    return responses_list
+
+
+def parse_response(text):
+    # Special cases
+    if '(broken file)' in text:
+        return ''
+    if 'versus (TI ' in text:
+        return ''
+    if 'Ceeeb' in text:
+        return ''
+
+    text = re.sub(r'…', '...', text)  # Replace ellipsis with three dots
+
+    regexps_empty_sub = [r'<!--.*?-->',  # Remove comments
+                         r'{{resp\|(r|u|\d+|d\|\d+)}}',  # Remove response rarity
+                         r'{{hero icon\|[a-z- \']+\|\d+px}}',  # Remove hero icon
+                         r'{{item( icon)?\|[a-z0-9() \']+\|\d+px}}',  # Remove item icon
+                         r'\[\[File:[a-z.,!\'() ]+\|\d+px\|link=[a-z,!\'() ]+]]',  # Remove Files
+                         r'<small>\[\[#[a-z_\-\' ]+\|\'\'followup\'\']]</small>',
+                         # Remove followup links in <small> tags
+                         r'<small>\'\'[a-z0-9 /]+\'\'</small>',  # Remove text in <small> tags
+                         r'<ref>.*?</ref>',  # Remove text in <ref> tags
+                         r'<nowiki>.*?</nowiki>',  # Remove text in <nowiki> tags
+                         ]
+    for regex in regexps_empty_sub:
+        text = re.sub(regex, '', text, flags=re.IGNORECASE)
+
+    regexps_sub_text = [r'\[\[([a-zé().:\',\- ]+)]]',  # Replace links such as [[Shitty Wizard]]
+                        r'\[\[[a-zé0-9().:\'/ ]+\|([a-zé().:\' ]+)]]',
+                        # Replace links such as  [[Ancient (Building)|Ancients]] and [[:File:Axe|Axe]]
+                        # r'{{h\|([a-zé().:\' ]+)}}',  # Replace hero names
```

Comment:
Why is this one commented out? A TODO or a comment is necessary, otherwise it should be removed.

Reply:
There was a single instance in responses source where this template was used. I've fixed it in the gamepedia sources since then, should not be needed in future hopefully.